01PITT_INST-ULS_1_5/js/custom.js

```diff
@@ -267,45 +275,170 @@ angular.module('hathiTrustAvailability', []).constant('hathiTrustBaseUrl', 'http
 });
 
 
-angular.module('getTempAddress', []).controller('prmRequestAfterController',function($scope){
-  //watch for the user to select a pickup location in the hold-request form
-  $scope.$watch(angular.bind($scope.$parent.$ctrl.requestService, function () {
-  if (typeof $scope.$parent.$ctrl.requestService !== 'undefined' && typeof $scope.$parent.$ctrl.requestService._formData !== 'undefined' && typeof $scope.$parent.$ctrl.requestService._formData.pickupLocation !== 'undefined'){
-    //result becomes newVal  
-    return $scope.$parent.$ctrl.requestService._formData.pickupLocation;
-   }
-   //every time the form changes:
-  }), function (newVal) {
-    //fix unecessary gap in instructions area:
-    //makes sure the form is loaded..
-    if (typeof angular.element(document.getElementsByTagName('prm-form-field')) !== 'undefined' && typeof angular.element(document.getElementsByTagName('prm-form-field'))[2] !== 'undefined' && typeof angular.element(document.getElementsByTagName('prm-form-field'))[2].children !== 'undefined'){
-      angular.element(document.getElementsByTagName('prm-form-field'))[2].children[0].children[0].style.marginTop='-25px';
+angular
+  .module('addressSelector', [])
+  .constant('addressServiceBaseUrl', 'https://dev-patron.libraries.pitt.edu/')
+  .controller('addressSelectorController', ['$scope', '$http', 'addressServiceBaseUrl', function($scope, $http, addressServiceBaseUrl) {
+    var self = this;
+
+    this.$onInit = function() {
+      $scope.getAddresses();
+    };
+
+    this.getJwt = function() {
+      var jwt = self.primoExplore.storageutil.sessionStorage.primoExploreJwt;
+      // strip quotes from jwt
+      if (jwt.charAt(0) === '"' && jwt.charAt(jwt.length - 1) === '"') {
+        jwt = jwt.slice(1, -1);
+      }    
+      return jwt;
+    }
+
+    $scope.addresses = [];
+    $scope.currentAddress = null;
+    $scope.showInput = false;
+    $scope.addressInput = {
+      line1: "",
+      line2: null,
+      line3: null,
+      line4: null,
+      line5: null,
+      city: "",
+      state_province: "",
+      postal_code: "",
+    };
+    $scope.statusMessage = "";
+    $scope.validationMessage = "";
+    $scope.shownAddressLines = 1;
+
+    $scope.showAddresses = function() {
+      return self.parentCtrl.requestService?._formData?.pickupLocation?.includes('$$USER_HOME_ADDRESS');
+    };
+
+    $scope.getAddresses = function() {
+      $scope.statusMessage = 'Loading address information...';
```

Comment:
there is a functional css-based loading animation in the sms module for use here